app/editor/src/components/form/datepicker/SelectDate.tsx

```diff
@@ -33,19 +34,41 @@ export const SelectDate: React.FC<IDatePickerProps> = ({
   tooltip,
   children,
   className,
+  initialDate,
   ...rest
 }) => {
+  const [startDate, setStartDate] = useState<Date>(new Date());
+  useEffect(() => {
+    if (initialDate) {
+      setStartDate(new Date(initialDate));
+    }
+  }, [initialDate]);
   return (
     <styled.SelectDate className="frm-in">
       {label && <label htmlFor={id ?? `txt-${name}`}>{label}</label>}
-      <DatePicker
-        name={name}
-        id={id}
-        className={`dpk ${className ?? ''}`}
-        data-for="main-tooltip"
-        data-tip={tooltip}
-        {...rest}
-      />
+      {!!initialDate ? (
```

Comment:
Based off the docs a state variable is supposed to be provided to `selected` and its setter to `onChange`, so when the form is loading a value, initialDate can be passed to our component to determine the appropriate selected. But looking at this now this can be done in a cleaner way. Will update.

Reply:
Some unfinished changes on the conditional rendering as well

Reply:
Tried to control these from the parent component to clean things up; however, react-datepicker seems to want to have the state controlled directly against it's base component.